src/System.Runtime.Extensions/src/System/ApplicationId.cs

```diff
@@ -41,7 +41,8 @@ public ApplicationId(byte[] publicKeyToken, string name, Version version, string
 
         public override string ToString ()
         {
-            StringBuilder sb = StringBuilderCache.Acquire();
+            Span<char> charSpan = stackalloc char[255];
```

Comment:
What's a typical length of these ApplicationIds? 255 is fine, but if they're all going to be much less than that, it'd be nice to save on the stack space by using 128 or something smaller (but only if we think that the 99% case will fit into that smaller size).

Reply:
`culture="en-us", version="1.0.0.0", publicKeyToken="43cb1e8e7a352766", processorArchitecture="x86"`
has 98 chars, leaving 30 chars for application name. I don't have a statictics for application names, but I met already 38 char exe name in wild. I'm rather go to 255. What's your decision?

Reply:
Works for me. thanks!

Reply:
> I met already 38 char exe name in wild

As Stephen said, this is about optimizing for the 99% case. I think that it is safe to say that 99% of the .exes out there have name shorter than 38 chars.

Reply:
done